src/browser/api/application.js

```diff
@@ -747,17 +749,24 @@ function run(identity, mainWindowOpts, userAppConfigArgs) {
 }
 
 /**
- * Run an application via RVM
+ * Run an application via RVM. Returns a promise which never resolves, only rejects.
```

Comment:
Lets hold off on this one till your back. Though its little mem-wise, we should find a way to have this not hang/leak on success and be a bit more intuitive (success is the clean path, not reject)